### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by AndrewHudzenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,8 +1,56 @@
...
+                builder.append(currentLine).append(System.lineSeparator());
+                currentLine = reader.readLine();
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Can't read file: " + fromFileName, e);
+        }
+        return builder.toString().split(System.lineSeparator());
+    }
+
+    private String totalStatCalculation(String[] splitStrings) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:29]:
```suggestion
    private String calculateResult(String[] splitStrings) {
```
use verbs for method names

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,8 +1,56 @@
...
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
 public class WorkWithFile {
 
     public void getStatistic(String fromFileName, String toFileName) {
+        readDataFromFile(fromFileName);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:11]: why is result ignored?

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,8 +1,56 @@
...
+    private String totalStatCalculation(String[] splitStrings) {
+        int supplySum = 0;
+        int buySum = 0;
+        for (String currentString : splitStrings) {
+            String[] twoValues = currentString.split(",");
+            if (twoValues[0].equals("supply")) {
+                supplySum += Integer.parseInt(twoValues[1]);
+            }
+            if (twoValues[0].equals("buy")) {
+                buySum += Integer.parseInt(twoValues[1]);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:38]: move delimiter and indices to constants

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,8 +1,56 @@
...
+        }
+        int result = supplySum - buySum;
+
+        return new StringBuilder().append("supply,")
+                .append(supplySum).append(System.lineSeparator())
+                .append("buy,").append(buySum).append(System.lineSeparator())
+                .append("result,").append(result).toString();
+    }
 
+    private void writeDataToFile(String toFileName, String totalStatCalculation) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:49]:
```suggestion
    private void writeDataToFile(String toFileName, String result) {
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -6,10 +6,16 @@
...
+    public static final String COMMA = ",";
+    public static final String SUPPLY = "supply";
+    public static final String BUY = "buy";
+    public static final int FIRST_INDEX = 0;
+    public static final int SECOND_INDEX = 1;
 
     public void getStatistic(String fromFileName, String toFileName) {
         readDataFromFile(fromFileName);
-        writeDataToFile(toFileName, totalStatCalculation(readDataFromFile(fromFileName)));
+        calculateResult(readDataFromFile(fromFileName));
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:17]: again - the result of calling this method is ignored

please look through these lines attentively

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 5 above)
Comment (AndrewHudzenko) [reply src/main/java/core/basesyntax/WorkWithFile.java:17]: ok , i will try it again ,but I don't  completely understand what you mean(

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,8 +1,62 @@
...
+    public static final String COMMA = ",";
+    public static final String SUPPLY = "supply";
+    public static final String BUY = "buy";
+    public static final int FIRST_INDEX = 0;
+    public static final int SECOND_INDEX = 1;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] lines = readDataFromFile(fromFileName);
+        String statistic = calculateResult(lines);
+        writeDataToFile(toFileName,statistic);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:18]: now everything is good

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 7]
```diff
@@ -1,8 +1,62 @@
...
+
+    private String[] readDataFromFile(String fromFileName) {
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String currentLine = reader.readLine();
+            while (currentLine != null) {
+                builder.append(currentLine).append(System.lineSeparator());
+                currentLine = reader.readLine();
+            }
+        } catch (Exception e) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:29]: sorry, 1 more thing
let's catch IOException here instead

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 8]
```diff
@@ -1,8 +1,62 @@
...
+        return new StringBuilder().append(SUPPLY).append(COMMA)
+                .append(supplySum).append(System.lineSeparator())
+                .append(BUY).append(COMMA).append(buySum).append(System.lineSeparator())
+                .append("result").append(COMMA).append(result).toString();
+    }
 
+    private void writeDataToFile(String toFileName, String result) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName, true))) {
+            writer.write(result);
+        } catch (Exception e) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:58]: same
we should aim to catch the narrowest exception

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**